src/Chainweb/CutDB.hs

```diff
@@ -407,15 +408,13 @@ startCutDb config logfun headerStore payloadStore cutHashesStore = mask_ $ do
                     tableIterPrev it
                     go it
                 Left e -> throwM e
-                Right hm -> do
-                    hm' <- case _cutDbParamsInitialHeightLimit config of
+                Right hm -> unsafeMkCut v
+                    <$> case _cutDbParamsInitialHeightLimit config of
                         Nothing -> return hm
                         Just h -> limitCutHeaders wbhdb h hm
-                    logg Debug $ "joining intial cut from cut db configuration with cut that was loaded from the database " <> sshow hm'
-                    joinIntoHeavier_
-                        (_webBlockHeaderStoreCas headerStore)
-                        hm'
-                        (_cutMap $ _cutDbParamsInitialCut config)
```

Comment:
This was joining the persisted cut with the genesis cut. By joining the cuts, it was validated that the stored cut was actually a cut (correctly braided) and all headers in the history of the cut exist in the chain db.

The algorithm for joining cuts creates a heap the includes all headers in the symmetric difference of the involved cuts. In the case of the join with the genesis cut, this is all headers in the history of the chain, which at the current block height in mainnet requires > 8GB. Nodes with <=8GB would run out of memory and crash.

By removing this step we just trust the stored cut in the DB. Which should be fine, since we trust the db in any case.

There are more efficient (and more complete) ways to validate the integrity of the DB, which can be performed offline. It would also be possible to add support to do this in the background in a chainweb node. But that's beyond the scope of the PR.

Reply:
The old check would seem not just to be RAM intensive but probably take a while, no?

Also is it truly redundant? Are there any modes where the check would alert of a corrupted node?

Reply:
The check would capture some database corruptions but not all. I see little value in it. In particular, it the failure is just an invalid braiding, it would be corrected as soon as the next remote cut is received. Only, if all nodes would have invalidly braided cuts it would be an issue.

If a block header is missing, the node would either work normally, if the header isn't used or it would fail if it tries to access the header. Depending on the situation the node may or may not recover from it by querying the missing header from a remote node.

Reply:
On nodes with sufficient RAM, the test is surprisingly fast -- still it takes some time. The runtime seems to scales super-linear under RAM pressure, probably due to caching issues.

However, DB pruning seems to be the main runtime bottleneck during startup. In particular on nodes with low number of cores (since it's concurrent for chains).

Reply:
I asked about time, b/c if nodes were taking this time, we could probably design a memory-efficient iterative version of this check, but it sounds like a different/more exhaustive check (and probably offline) is better.